File: decoder/geoutils/maths.py
"""Small math helpers used by geoutils."""

import numpy as np


class Maths:
    """Small math utility functions."""

    @staticmethod
    def hypot(a: float, b: float) -> float:
        """Stable hypot implementation (avoid overflow/underflow)."""
        if abs(a) > abs(b):
            r = b / a
            return abs(a) * np.sqrt(1 + r * r)
        elif b != 0:
            r = a / b
            return abs(b) * np.sqrt(1 + r * r)
        else:
            return 0.0

File: PGTA Source/PGTA_deco048/Maths.cs
using System;

namespace PGTA_deco048;

internal class Maths
{
	public static double Hypot(double a, double b)
	{
		if (Math.Abs(a) > Math.Abs(b))
		{
			double num = b / a;
			return Math.Abs(a) * Math.Sqrt(1.0 + num * num);
		}
		if (b != 0.0)
		{
			double num = a / b;
			return Math.Abs(b) * Math.Sqrt(1.0 + num * num);
		}
		return 0.0;
	}
}
